Reader: Fix various card spacing inconsistencies

jancavan opened the pull request:
This fixes: https://github.com/Automattic/wp-calypso/issues/9144

Make sure spacing is consistent in all the cards:
- Default
- Excerpt
- Photo
- Gallery
- Video

> In addition to the 20px thing, I'm also suggesting that we limit the gallery card images to 100px tall when in the narrow screen

It also seems like spacing changes depending on viewport so have to check those as well.

fraying commented:
Woot! Nice quick work, Jan!

fraying commented:
Noticed the runaway feed here.

![screen shot 2016-11-04 at 2 12 59 pm](https://cloud.githubusercontent.com/assets/24388/20022664/d9ab4a94-a298-11e6-96c2-2139bae875e7.png)

samouri commented on client/blocks/reader-post-card/index.jsx:
```diff
@@ -86,27 +86,27 @@ export default class RefreshPostCard extends React.Component {
 		const isPhotoOnly = !! ( post.display_type & DisplayTypes.PHOTO_ONLY );
 		const isGallery = !! ( post.display_type & DisplayTypes.GALLERY );
-		const featuredAsset = ( <FeaturedAsset post={ post } /> );
+		const featuredAsset = FeaturedAsset( { post } ); // this is ... gross? gross.
```
what is gross about this?

blowery replied:
it's using a "component" as something other than a component. If we want to walk this way, it should be `buildFeaturedAsset` instead of `FeaturedAsset`.

jancavan commented:
In https://github.com/Automattic/wp-calypso/pull/9163/commits/64e2c750b99fbc40c37a5e439283176fdd582af7:

## **Default card:**
![default-1](https://cloud.githubusercontent.com/assets/4924246/20118291/bfabd956-a5b8-11e6-9dff-89e6ccf54185.jpg)
![default-2](https://cloud.githubusercontent.com/assets/4924246/20118293/bfaeb4aa-a5b8-11e6-9963-bc3f03cbca9d.jpg)
![default-3](https://cloud.githubusercontent.com/assets/4924246/20118292/bfae3584-a5b8-11e6-8ae7-2b23a3d2bc76.jpg)

## **Excerpt/text-only card:**
![excerpt-1](https://cloud.githubusercontent.com/assets/4924246/20118322/d854832c-a5b8-11e6-97cf-3fc1f8652e8d.jpg)
![excerpt-2](https://cloud.githubusercontent.com/assets/4924246/20118321/d85381ac-a5b8-11e6-948c-95fba079701d.jpg)

## **Photo card:**
![photo-1](https://cloud.githubusercontent.com/assets/4924246/20118355/f3461fe2-a5b8-11e6-925b-b85e3310da44.jpg)
![photo-2](https://cloud.githubusercontent.com/assets/4924246/20118356/f3483c28-a5b8-11e6-8af5-9f280c278a92.jpg)

## **Gallery card:**
![gallery-1](https://cloud.githubusercontent.com/assets/4924246/20118363/f98c333c-a5b8-11e6-9ee2-92f81e3e7ebb.jpg)
![gallery-2](https://cloud.githubusercontent.com/assets/4924246/20118364/f99c84da-a5b8-11e6-9192-e890c5277741.jpg)
![gallery-3](https://cloud.githubusercontent.com/assets/4924246/20118365/f9a42348-a5b8-11e6-87ed-0e9e6791dba8.jpg)

## **Video card:**
![video-1](https://cloud.githubusercontent.com/assets/4924246/20118351/eb4e34f0-a5b8-11e6-84cc-6df97b0040c7.jpg)
![video-2](https://cloud.githubusercontent.com/assets/4924246/20118350/eb4c3eac-a5b8-11e6-9546-5ec6a99bf5e3.jpg)

## **Video card (playing):**
![videoplay-1](https://cloud.githubusercontent.com/assets/4924246/20118332/e0ebc4f0-a5b8-11e6-8f0d-f70995d5d509.jpg)
![videoplay-2](https://cloud.githubusercontent.com/assets/4924246/20118333/e0ecf74e-a5b8-11e6-8bb9-9ae2a756148a.jpg)

- [ ] Check spacing in cards without gravs/blavs
- [ ] Make sure height of featured assets are correct in different viewports

fraying commented:
BEST. COMMIT. EVER.

Thank you so much, @jancavan. This is excellent.

jancavan commented:
In https://github.com/Automattic/wp-calypso/pull/9163/commits/013e17d67a296551343fe02f39b58e7980566ad1:

- Fixed gradient overlay in Safari for photo cards:

**Before:**
![screenshot 2016-11-09 12 23 56](https://cloud.githubusercontent.com/assets/4924246/20153411/8231628e-a677-11e6-9fbb-4b03ba14ebca.png)

**After:**
![screenshot 2016-11-09 12 24 06](https://cloud.githubusercontent.com/assets/4924246/20153420/85731cbc-a677-11e6-8959-78ef03a37357.png)

- Fixed misaligned thumbnail and PostOptions menu:

**Before:**
![screenshot 2016-11-09 12 27 41](https://cloud.githubusercontent.com/assets/4924246/20153575/402374ee-a678-11e6-890a-502c58ca68cf.png)

**After:**
![screenshot 2016-11-09 12 29 17](https://cloud.githubusercontent.com/assets/4924246/20153582/466bc270-a678-11e6-9fe4-844b5dcd0dc4.png)

> Check spacing in cards without gravs/blavs

Had a brief chat with @fraying. If there isn't a grav, spacing in between meta info and thumbnail/excerpt is `22px` instead of `20px`, but we both agreed it's fine since it won't be worth adding a separate class/styles just to fix this.

fraying commented:
🎉
